docs/design/virtual-addresses.md

```diff
@@ -0,0 +1,123 @@
+  - Accounts to utilize virtual addresses on creation
+    - EOAs set virtual address for regular `CryptoCreate` with ECDSA key, auto-create with alias and lazy-crate with `evmAddress`
+    - Contracts set virtual address for Smart contract `new()` (`CREATE` & `CREATE2`) and `EthereumTransaction CREATE` with empty `to` address
+    - Creation logic verifies `evmAddress` uniqueness and fails/doesn't set if entry already exists
+- Phase 3
+  - Protobuf changes
+    - Add disable to `CryptoUpdateTransactionBody.virtual_address_update`
+  - Expand virtual address limit per account to 3
+  - Virtual address addition to existing accounts with signature verification logic and verification for `evmAddress` uniqueness
+  - Support `virtual_address_override` logic for `ContractCreate` and `ContractCall` transaction
+- Phase 4
+  - Support virtual address disabling on `CryptoUpdate`
+  - Design virtual address deletion & transfer
+
+## Acceptance Tests
+
+### Positive Tests
+- `CryptoCreate` with `ECDSA key` should create an account with single virtual address and make it the default virtual address
+- `CryptoCreate` with `ECDSA key alias` should create an account with single virtual address and make it the default virtual address
+- `CryptoCreate` with `evmAddress` should create a hollow account with single virtual address and make it the default virtual address
+- `CryptoTransfer` with `ECDSA key alias` to a non-existing account should auto-create an account with single virtual address and make it the default virtual address
+- `CryptoTransfer` with `evmAddress` to a non-existing account should lazy-create a hollow account with single virtual address and make it the default virtual address
+- `EthereumTransaction` to a non-existing account with `tx.to` EVM address value should lazy-create a hollow account with single virtual address and make it the default virtual address
+- `ContractCreate/ContractCall` for an account with a default virtual address should use that address in the EVM
+- `ContractCreate/ContractCall` with `virtual_address_override` address value should use that address in the EVM
+- `ContractCreate/ContractCall` resulting in creation of a new contract should add the CREATE/CREATE2 EVM address value to `contract.account.virtualAddresses`
+- `CryptoUpdate` with `virtual_address_update.add.address` for an existing account should add a new virtual address, if `virtual_address_update.add.is_default` is set to `true` the added address should become the default virtual address
+- `CryptoUpdate` with `virtual_address_update.disable` value that is present in the virtual address list for an existing account should disable the virtual address
+- `CryptoUpdate` with `virtual_address_update.remove` value that is present in the virtual address list for an existing account should remove the virtual address from the list
+- `CryptoGetInfoQuery` for an existing account should return the virtual addresses list for the account
+
+### Negative Tests
```

Comment:
Add migration tests to consider the following entity types.
- simple ED account
- ED account w alias
- simple EC account
- EC account with alias
- hollow account
- contract created via CREATE
- contract created via CREATE2

Also consider a test class to support automated migration testing that verify ledger and in memory maps before and after as an integration test

Reply:
Added separate section for migration tests in the test plan doc.